sieve.py

```diff
@@ -574,6 +573,36 @@ def run(
                 phase,
             )
 
+            result_map = {
+                "duration": time.time() - s,
+                "alarm": alarm,
+                "bug_report": bug_report,
+            }
+
+            # Testing mode, write test result under sieve_test_result directory
+            result_filename = "sieve_test_results/{}-{}-{}.json".format(project, test, os.path.basename(test_config))
+            if not os.path.exists(os.path.dirname(result_filename)):
+                try:
+                    os.makedirs(os.path.dirname(result_filename))
+                except OSError as exception:
+                    if exception.errno != errno.EEXIST:
+                        raise
+
+            with open(result_filename, 'w') as test_result_json:
+                json.dump(
+                    result_map,
+                    test_result_json,
+                    indent=4,
+                )
+
+            # XXX: Temporarily copy log information to save it for later analysis
+            test_log_save = os.path.join("log_save", project, test, stage)
```

Comment:
Can you append the current datetime to `log_save`? So that it will never be overwritten

Reply:
The idea is that each test run has a unique project/test/config (note I used test_config on line 602) tuple, since this path is unique between test runs, the results will not get overwritten. I didn't want to simply copy the log_save dir every test run because we would have hundreds of log dirs after finishing just the yugabyte-operator.

If we want to retain this log dir between each massive testing, I should add a script to fetch these log_save dirs back to master after finishing the and then append a timestamp to its name.

Reply:
I see. I misunderstood your code. LGTM